sdk/src/main/scala/io/horizen/certificatesubmitter/strategies/WithKeyRotationCircuitStrategy.scala

```diff
@@ -85,6 +94,7 @@ class WithKeyRotationCircuitStrategy[
       sidechainId,
       backwardTransfers,
       endEpochCumCommTreeHash,
+      sc2ScDataForCertificate,
```

Comment:
`CertificateDataWithKeyRotation` creation must be refactored.
This code in general will not work.
Couple of suggestions:
1) `Sc2ScDataForCertificate` can be removed at all
2) `Sc2ScDataForCertificate.previousTopQualityCertificateHash` can be retrieved from `previousCertificateOption`
3) `CertificateDataWithKeyRotation` custom fields must be constructed using circuit specific custom fields appended with sc2sc if needed (if feature is enabled AFTER fork activation only).